SMTPServer/MailSession.cpp

```diff
@@ -0,0 +1,337 @@
+// This is an independent project of an individual developer. Dear PVS-Studio, please check it.
+
+// PVS-Studio Static Code Analyzer for C, C++, C#, and Java: http://www.viva64.com
+
+#include "MailSession.h"
+
+MailSession::MailSession(SOCKET client_socket)
+{
+	this->m_client_socket = client_socket;
+}
+
+const SOCKET& MailSession::get_client_socket() const
+{
+	return m_client_socket;
+}
+
+bool MailSession::ValidAdress(char* buf)
+{
+	size_t strlen_buf = strlen(buf);
+	return (strlen_buf > MIN_MAIL_SIZE && strlen_buf < MAX_MAIL_SIZE && strchr(buf, '@'));
+}
+
+std::string MailSession::CutAddress(char* buf)
+{
+	std::string str_buf = buf;
+	std::string result = "";
+
+	size_t start_pos = str_buf.find('<', 0);
+	size_t end_pos = str_buf.find('>', 0);
+
+	for (size_t i = start_pos + 1; i < end_pos; i++)
+	{
+		result += str_buf.at(i);
+	}
+
+	return result;
+}
+
+std::string MailSession::CutSubject(char* buf)
+{
+	std::string str_buf = buf;
+	std::string result = "";
+
+	size_t start_pos = str_buf.find(':', 0);
+
+	for (size_t i = start_pos + 2; i < str_buf.size(); i++)
+	{
+		result += str_buf.at(i);
+	}
+
+	return result;
+}
+
+int MailSession::SendResponse(int response_type)
+{
+	char buf[RESPONSE_BUF_SIZE];
+	ZeroMemory(&buf, sizeof(buf));
+
+	if (response_type == Responses::WELCOME_TO_CLIENT)
+	{
+		strcpy(buf, "220 Welcome!\r\n");
+	}
+
+	else if (response_type == Responses::SERVICE_CLOSING)
+	{
+		strcpy(buf, "221 Service closing transmission channel\r\n");
```

Comment:
Please create const string variable for "221 Service closing transmission channel\r\n"